### PR TITLE
Adjust crop rect if outside of image boundaries

### PR DESCRIPTION
**Pull request by sammy-SC**

### CODE REVIEW

**Comment by sammy-SC** [comment]
[Crop parameters for cover images can be invalid when set in client](https://trello.com/c/87qrFdLe/625-crop-parameters-for-cover-images-can-be-invalid-when-set-in-client)

**Review by famanson — APPROVED** [review]
@sammy-SC approved

File: TOCropViewController/Views/TOCropView.m [hunk 1]
```diff
@@ -819,6 +819,16 @@ - (CGRect)croppedImageFrame
     frame.size.height = ceilf(cropBoxFrame.size.height * (imageSize.height / contentSize.height));
     frame.size.height = MIN(imageSize.height, frame.size.height);
     
+    // if frame goes beyond boundaries of the image, we move it back
+    // so it is within the boundaries.
+    if (frame.origin.x + frame.size.width > imageSize.width) {
+        frame.origin.x -= frame.origin.x + frame.size.width - imageSize.width;
```
Comment (famanson) [inline TOCropViewController/Views/TOCropView.m]: @sammy-SC actually, upon reading these lines again, you can do:

```
frame.origin.x = imageSize.width - frame.size.width;
```

And it has the same effect

File: TOCropViewController/Views/TOCropView.m [hunk 2]
```diff
@@ -819,6 +819,16 @@ - (CGRect)croppedImageFrame
...
+    }
+
+    if (frame.origin.y + frame.size.height > imageSize.height) {
+        frame.origin.y -= frame.origin.y + frame.size.height - imageSize.height;
```
Comment (famanson) [inline TOCropViewController/Views/TOCropView.m]: @sammy-SC same here

**Comment by sammy-SC** [comment]
@famanson thanks, changed.

**Comment by famanson** [comment]
@sammy-SC please test it yourself unless you have already done so, remove the `invalid` label then merge

**Comment by sammy-SC** [comment]
@famanson it was tested even in the app, works as expected.

**Comment by famanson** [comment]
@sammy-SC merged

**Pull request merged**